WordPress/gutenberg
# Fix the block toolbar styles to evenly space buttons

Follow-up to #39605

## What?

In https://github.com/WordPress/gutenberg/pull/39605#issuecomment-1073644254 we noticed that the block toolbar spacing is not correct when the block toolbar contains two buttons with descriptions. The descriptions show hidden divs in the DOM impacting the `:first-child` `last-child` selectors that are supposed to apply the correct styles to the button toolbars.

## How?

This PR solves that by removing the special cases from the "first" and "last" child buttons of block toolbars. Instead the containing toolbar group now has a 6px padding. The impact here is that the click area of the first and last buttons of a "toolbar group" is a bit smaller but it's also more consistent with all the remaining toolbar buttons.

## Testing Instructions

1- Use different blocks and toolbars
2- make sure the toolbar buttons are evenly spaced.

## Screenshots or screencast 

I added a red outline to the impacted buttons in the screenshots here

<img width="453" alt="Screen Shot 2022-03-22 at 9 49 42 AM" src="https://user-images.githubusercontent.com/272444/159442591-7c1d7fd2-d175-4cd9-a587-62c74bc93930.png">

<img width="381" alt="Screen Shot 2022-03-22 at 9 49 48 AM" src="https://user-images.githubusercontent.com/272444/159442628-42d03215-5c6f-4aaa-9830-7c75069771ee.png">

<img width="96" alt="Screen Shot 2022-03-22 at 9 50 00 AM" src="https://user-images.githubusercontent.com/272444/159442696-75822bb8-b61d-48f6-8f58-7f56787c4b9f.png">
